### PR TITLE
Bring back creating non-native installers

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by mcg1969** [comment]
I'm 100% OK with this but it's not quite done yet if I remember correctly—specifically, we'd need to prevent an attempt to build Windows installers on Mac and Linux, correct?

**Comment by mcg1969** [comment]
Also, I was having significant problems actually getting this to work. Conda continued to insist on downloading native packages.

**Comment by isuruf** [comment]
Not really. NSIS can be run on Linux or Mac although there's no conda packages for them yet.

File: constructor/fcp.py [hunk 1]
```diff
@@ -319,7 +327,7 @@ def main(info, verbose=True, dry_run=False, conda_exe="conda.exe"):
     version = info["version"]
     download_dir = info["_download_dir"]
     platform = info["_platform"]
-    channel_urls = all_channel_urls(info.get("channels", ()))
+    channel_urls = all_channel_urls(info.get("channels", ()), subdirs=[platform, "noarch"])
```
Comment (isuruf) [inline constructor/fcp.py:330]: @mcg1969, this fixed the issue where conda was trying to use native packages.

**Comment by mcg1969** [comment]
So before we merge this, can you verify that both building Linux installers on the Mac and building Mac installers on Linux results in properly functioning installers?

**Comment by mcg1969** [comment]
Nevermind my last comment, I see the change you highlighted. All good there.

But back to Windows builds on Linux. If it's true that we can run NSIS on Mac and/or Linux, great. However, it doesn't seem reasonable to me that we make it a required dependency for constructor on those platforms. So at the very least we would need some mechanism to detect the existence of NSIS on those platforms, and error out if it doesn't exist. Kind of like a package that has an optional conda dependency, providing helpful messaging when the package isn't installed.

**Comment by isuruf** [comment]
I've tested that the Linux installer on Mac works correctly. I'll try the other way too

**Comment by isuruf** [comment]
About windows build on Linux, there's a good error message already at https://github.com/conda/constructor/blob/e81be544bfcc9abb98459ebdf83cb602a1bc8f6d/constructor/winexe.py#L135

**Comment by mcg1969** [comment]
OK, thank you for addressing my concerns! And sorry about this. It seems clear that I misinterpreted things. When the cross-platform builds will failing for us, I got the false impression that they had been broken for awhile.

**Comment by isuruf** [comment]
Thanks

**Comment by mcg1969** [comment]
Oops, we broke the `environment` option with this, because of where you moved the `user_conda` determination

File: constructor/fcp.py [hunk 2]
```diff
@@ -250,19 +251,26 @@ def _main(name, version, download_dir, platform, channel_urls=(), channels_remap
...
     # make the environment, if needed
     if environment_file:
+        # set conda to be the user's conda (what is in the environment)
+        # for purposese of getting & building environements, rather
+        # than the standalone conda (conda_exe). Fallback to the
+        # standalone, if needed
+        user_conda = os.environ.get('CONDA_EXE', '')
+        if not user_conda:
+            if cc_platform == platform:
+                conda_exe
```
Comment (pelson) [inline constructor/fcp.py:263]: Woops. This breaks being able to use constructor without having activated conda. I guess you meant ``user_conda = conda_exe``?

Comment (isuruf) [reply constructor/fcp.py:263]: Yes, I meant `user_conda = conda_exe`. Sorry about the bug.

Comment (pelson) [reply constructor/fcp.py:263]: No worries. Fixed in #422.

**Pull request merged**